### ISSUE TITLE
APIGatewayClient fail to authenticate when identityId is part of the rest api path. Solution provided!

### ISSUE THREAD

**Issue by ChuckZoo** [aws-amplify/aws-sdk-ios#1289]
**Describe the bug**
I found this issue when I send request to my APIGateway API.
APIGateway Path: users/{identityId}/info
Expected URL: users/us-east-1:6e79ea21-4ed9-45cf-8037-de5129163b42/info
Actual URL: us-east-1%3A6e79ea21-4ed9-45cf-8037-de5129163b42

Problematic Method:

- (AWSTask<AWSAPIGatewayResponse *> *)invoke:(AWSAPIGatewayRequest *)apiRequest

It encodes the identityId in the url path, which cause Cogito cannot recognize the identity.
After I removed that line of code that encode the URL, the problem fixed.

The path part should not be encoded any way, please fix this.

**Comment by rohandubal** [comment]
Hello @ChuckZoo 

Sorry that you are having issues using the SDK. This does seem like a URL encoding issue.

Could you please give some details so that we can debug and fix the issue?
- How are you using the API Gateway SDK? Are you using a generated SDK or raw invocation by passing all parameters?
- How do you pass the identity id to the request?
- Could you provide a sample code snippet by which we can reproduce this?

Thanks,
Rohan

**Comment by ChuckZoo** [comment]
Hi Rohan

- I have the backend by serverless framework, so I have to do the raw invocation instead of amplify generated.
- I passed the identity in the url path, so I can utilize AWS IAM to authorize. My API gateway path is like users/{id}/info, which id is the identity id
- Here's how to invoke the api. As you see below, the problem is that urlString I passed in gets encoded by the SDK.
`
let request = AWSAPIGatewayRequest(httpMethod: "POST",
                                              urlString: "users/us-east-1:d1751f08-0126-4cd9-89e1-29e894e4df18/info",
                                              queryParameters: queryParams,
                                              headerParameters: headers,
                                              httpBody: String(dictionary: body))
        
let client = AWSAPIGatewayClient()
client.configuration = serviceConfiguration()
client.invoke(request)

`
